native/common/jp_class.cpp

```diff
@@ -321,10 +322,29 @@ JPPyObject JPClass::convertToPythonObject(JPJavaFrame& frame, jvalue value, bool
 
 	if (isThrowable())
 	{
+		JPPyTuple tuple1 = JPPyTuple::newTuple(2);
+		tuple1.setItem(0, _JObjectKey);
+		if (value.l == NULL)
+		{
+			JPPyTuple tuple0 = JPPyTuple::newTuple(0);
+			tuple1.setItem(1, tuple0.get());
+		} else
+		{
+			JPPyTuple tuple0 = JPPyTuple::newTuple(1);
+			jstring m = frame.getMessage((jthrowable) value.l);
+			if (m != NULL)
+			{
+				tuple0.setItem(0, JPPyString::fromStringUTF8(
+						frame.toStringUTF8(m)).get());
+			} else
+			{
+				tuple0.setItem(0, JPPyString::fromStringUTF8(
+						frame.toString(value.l)).get());
+			}
+			tuple1.setItem(1, tuple0.get());
+		}
 		// Exceptions need new and init
```

Comment:
We get a new block of code that works out what the proper arguments to an exception would have been had it be thrown from Python.  Thus Python and Java thrown exceptions are now identical.